src/pip/_internal/req/constructors.py

```diff
@@ -115,23 +115,17 @@ def parse_editable(editable_req):
             url = f'{version_control}+{url}'
             break
 
-    if '+' not in url:
+    link = Link(url)
+
+    if not link.is_vcs:
+        backends = ", ".join(f"{backend.name}+" for backend in vcs.backends)
         raise InstallationError(
-            '{} is not a valid editable requirement. '
-            'It should either be a path to a local project or a VCS URL '
-            '(beginning with svn+, git+, hg+, or bzr+).'.format(editable_req)
+            f'{editable_req} is not a valid editable requirement. '
+            f'It should either be a path to a local project or a VCS URL '
+            f'(beginning with {backends}).'
```

Comment:
I'm wondering if we should not simply show `vcs.all_schemes` directly.
Otherwise, we might crash on someone using `git+ftp` (or something else) and tell him to use a VCS URL beginning with `git+` which is already the case.
Yes, it seems far fetched ^^

Reply:
Two possible issues with that are
1/ the list of all schemes is very long
2/ it still contains `git`, `svn`, etc (without `+`)

Regarding 2/, VCS schemes without `+` such as `git://` are supported in editable mode only (see a few lines above), and I think it's not something we want to encourage. I suspect we could remove the schemes without `+` from `all_schemes` but I'm not 100% sure.

Reply:
@xavfernandez I've double checked and I don't see a reason for having the schemes without `+` in the list of all VCS schemes. So I removed them and did as you suggested to show the full list of supported schemes in the error message.